### ISSUE TITLE
False positive in rule: function-paren-newline when using type parameter

### ISSUE THREAD

**Issue by levchak0910** [eslint-stylistic/eslint-stylistic#206]
### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

In several scenarios first opening paren is found in type parameters, which leads to falsy error and incorrect fix

Examples:

- function declaration

```ts
function a<A extends Array<any>, T extends (...args: any[]) => any> (
  b: T,
  c: any,
): any {}
```

- function expression

```ts
const a = function <A extends Array<any>, T extends (...args: any[]) => any> (
  b: T,
  c: any,
): any {}
```

- call expression

```ts
a<Array<any>, (...args: any[]) => any>(
  b,
  c,
)
```

### Reproduction

in description

### Contributes

- [X] I am willing to submit a PR to fix this issue
- [X] I am willing to submit a PR with failing tests
